services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,8 +223,6 @@ public void run()
             )
         );
         injector.getInstance(ExecutorLifecycle.class).join();
-        // Explicitly call lifecycle stop, dont rely on shutdown hook.
-        lifecycle.stop();
```

Comment:
It was discussed in the dev-sync that @himanshug 's approach is probably the way to go for now. This may require changes to Lifecycle in java-util such that stop can be applied multiple times. But for the purposes of this PR I'll revert the behavior here.